### PR TITLE
Bump to v5.0.0-RC7

### PR DESCRIPTION
**Pull request by mheon**
As the title says

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: version/rawversion/version.go
Comment (vrothberg) [inline version/rawversion/version.go:10]: Should this be 5.1.0-dev?

**Review by vrothberg — COMMENTED** [review]
LGTM, restarted flaked jobs

File: version/rawversion/version.go
Comment (lsm5) [reply version/rawversion/version.go:10]: nope, this is already on v5.0 branch

Comment (lsm5) [reply version/rawversion/version.go:10]: but ya, we need to bump main to 5.1.0-dev

Comment (lsm5) [reply version/rawversion/version.go:10]: https://github.com/containers/podman/pull/22054

**Comment by mheon** [comment]
@containers/podman-maintainers PTAL and merge

File: RELEASE_NOTES.md
Comment (Luap99) [inline RELEASE_NOTES.md:91]: should this be `Changed the ...`? It sounds odd without that, anyhow I am fine to merge to not wait for another CI cycle and fic that up for the final release.

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

File: RELEASE_NOTES.md
Comment (mheon) [reply RELEASE_NOTES.md:91]: Yeah, that sounds better

**Pull request merged**
